qiita_ware/test/test_private_plugin.py

```diff
@@ -413,7 +415,41 @@ def test_delete_study_empty_study(self):
 @qiita_test_checker()
 class TestPrivatePluginDeleteAnalysis(BaseTestPrivatePlugin):
     def test_delete_analysis(self):
-        # as samples have been submitted to EBI, this will fail
+        # adding extra filepaths to make sure the delete works as expected, we
+        # basically want 8 -> 9 -> 10 -> 12 -> 14
+        #                       -> 11 -> 13
+        fd, fp10 = mkstemp(suffix='_table.biom')
```

Comment:
Are these being added to a list of files to delete? If not, then maybe would be better to use TemporaryFile instead:

From the docs:
> Unlike TemporaryFile(), the user of mkstemp() is responsible for deleting the temporary file when done with it.

Reply:
Also, looks like `fd` needs to explicitly be closed. I seemed to recall that, and looking at other portions of code in this file seem to confirm it.